src/platform/i386/miniacpi.c

```diff
@@ -0,0 +1,108 @@
+#include "kernel.h"
+#include "string.h"
+#include "mem_layout.h"
+#include "pgtbl.h"
+
+typedef struct {
+	char signature[8];
+	u8_t checksum;
+	char oemid[6];
+	u8_t revision;
+	u32_t rsdtaddress;
+	u32_t length;
+	u64_t xsdtaddress;
+	u8_t extendedchecksum;
+	u8_t reserved[3];
+} __attribute__((packed)) RSDP;
+
+typedef struct rsdt {
+	char signature[4];
+	u32_t length;
+	u8_t revision;
+	u8_t checksum;
+	char oemid[6];
+	char oemtableid[8];
+	u32_t oemrevision;
+	u32_t creatorid;
+	u32_t creatorrevision;
+	struct rsdt *entry[0];
+} __attribute__((packed)) RSDT;
+
+extern u8_t *boot_comp_pgd;
+u32_t basepage;
```

Comment:
static?  Can it be accessed outside of this file?  Will it still be a u32_t on a 64 bit machine?